code/parachain/frame/composable-traits/src/storage.rs

```diff
@@ -0,0 +1,58 @@
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+
+/// Semantic abstraction of `Option` for updating storage items in monolithic interfaces where not
+/// all fields need updated.
+#[derive(Decode, Encode, Debug, Clone, PartialEq, Eq, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum UpdateValue<T> {
+	/// Value will **not** be set/updated - will remain the same.
+	DoNotSet,
+	/// Value will be set/updated - will be changed.
+	Set(T),
+}
+
+impl<T> UpdateValue<T> {
+	/// Returns the contained [`Set`] value or a default.
+	///
+	/// Consumes the `self` argument then, if [`Set`], returns the contained
+	/// value, otherwise if [`DoNotSet`], returns the [default value] for that
+	/// type.
+	///
+	/// # Examples
+	///
+	/// ```
+	/// let x: UpdateValue<u32> = DoNotSet;
+	/// let y: UpdateValue<u32> = Set(12);
+	///
+	/// assert_eq!(x.unwrap_or_default(), 0);
+	/// assert_eq!(y.unwrap_or_default(), 12);
+	/// ```
+	pub fn unwrap_or_default(self) -> T
```

Comment:
this is unused afaict, is it necessary to add right now?

Reply:
same with the From<Option> and Default impls, this type should be constructed literally whenever it's used imo

Reply:
This was designed to possible be used in other contexts as well. So I added these thinking they may be helpful in testing.